docs/machine-learning/basic-concepts-model-training-in-mldotnet.MD

```diff
@@ -0,0 +1,169 @@
+ms.date: 11/06/2018
+ms.topic: overview
+ms.custom: mvc
+#Customer intent: As a developer, I want to learn the basic ML.NET concepts related to model training so that I can understand how to leverage machine learning in my applications.
+---
+# Basic concepts for model training in ML.NET
+
+This article explains the basic concepts covering *model training* scenarios in ML.NET. Not all concepts are relevant to the more simple scenario of *prediction with existing model*.
+
+## Model training concepts
+
+This document covers the following concepts related to model training in ML.NET:
+
+- [*Data*](#data), represented as an `IDataView` interface.
+  - In ML.NET, data is very similar to a SQL view: it's a lazily-evaluated, immutable, cursorable, heterogenous, schematized dataset. 
+  - An excellent document about the data interface is [IDataView Design Principles](https://github.com/dotnet/machinelearning/blob/master/docs/code/IDataViewDesignPrinciples.md).
+- [*Transformer*](#transformer), represented as `ITransformer` interface.
+  - In one sentence, a transformer is a component that takes data, does some work on it, and returns new 'transformed' data.
+  - For example, you can think of a machine learning model as a transformer that takes features and returns predictions.
+  - Another example, 'text tokenizer' would take a single text column and output a vector column with individual 'words' extracted out of the texts.
+- [*Data reader*](#data-reader), represented as an `IDataReader<T>` interface.
+  - The data reader is ML.NET component to 'create' data: it takes an instance of `T` and returns data out of it. 
+  - For example, a *TextLoader* is an `IDataReader<FileSource>`: it takes the file source and produces data. 
+- [*Estimator*](#estimator), represented as an `IEstimator<T>` interface.
+  - This is an object that learns from data. The result of the learning is a *transformer*.
+  - You can think of a machine learning *algorithm* as an estimator that learns on data and produces a machine learning *model* (which is a transformer).
+- [*Prediction function*](#prediction-function), represented as a `PredictionFunction<TSrc, TDst>` class.
+  - The prediction function can be seen as a machine that applies a transformer to one 'row', such as at prediction time.
+
+## Data
+
+In ML.NET, data is very similar to a SQL view: it's a lazily-evaluated, cursorable, heterogenous, schematized dataset.
+
+- It has *Schema* (an instance of an `ISchema` interface), that contains the information about the data view's columns.
+  - Each column has a *Name*, a *Type*, and an arbitrary set of *metadata* associated with it.
+  - It is important to note that one of the types is the `vector<T, N>` type, which means that the column's values are *vectors of items of type T, with the size of N*. This is a recommended way to represent multi-dimensional data associated with every row, like pixels in an image, or tokens in a text.
```

Comment:
The notation of `vector<T, N>` is misleading. From the description, this isn't meant to a a .NET generic type with two type parameters. Instead, it reads like you intend this type to be a vector of type `T`, with a fixed size `N`. Why not just say it's a vector or an array?

Reply:
Deadline constraints; will be handled later.

Reply:
In the meantime, let's delete this paragraph. It's more confusing than enlightening.